src/main/app/controller/PostController.ts

```diff
@@ -18,7 +19,12 @@ export class PostController<T extends AnyObject> {
   public async post(req: AppRequest<T>, res: Response): Promise<void> {
     const { saveAndSignOut, _csrf, ...formData } = this.form.getParsedBody(req.body);
 
-    Object.assign(req.session.userCase, formData);
+    const { id, divorceOrDissolution } = req.session.userCase;
+    req.session.userCase = {
+      id,
+      divorceOrDissolution,
+      ...omitUnreachableAnswers({ ...req.session.userCase, ...formData }, req.app.locals.steps),
+    };
```

Comment:
what does this do?

Reply:
This omits unreachable answers on the check your answers page. It's based on the steps, their case and form state.

For example if they were to change "Did you get married in the UK" to Yes then it would remove these "Where did you get married" sub questions (if they had answered previously answer them) from the "Check you answers" page.

Reply:
I'll move this out to another PR, as technically this is not part of this ticket